### PR TITLE
implement support for resources

### PR DESCRIPTION
**Pull request by dicej**
This builds on Guy's earlier, unfinished work.  I've ported over all the resource tests I wrote for `componentize-py` and tried to make a minimal number of changes needed to get them passing.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,14 +16,14 @@ base64 = "0.21.2"
 bitflags = "1.3.2"
 env_logger = "0.10.0"
 heck =  { version = "0.4", features = ["unicode"] }
-js-component-bindgen = { git = "https://github.com/bytecodealliance/jco", branch = "main", no-default-features = ["transpile-bindgen"] }
+js-component-bindgen = { path = "../jco/crates/js-component-bindgen", no-default-features = ["transpile-bindgen"] }#{ git = "https://github.com/dicej/jco", branch = "resources", no-default-features = ["transpile-bindgen"] }
```
[the remark this answers is not in the document]
Comment (dicej) [reply Cargo.toml:19]: This part's easy enough, but I also need to use the latest jco in package.json.  Do we need to do a release for that?  I tried pointing package.json to the GitHub repo, but couldn't get it working.

**Comment by guybedford** [comment]
I've released jco@0.12.2, so we can try that here now.

**Comment by dicej** [comment]
> I've released jco@0.12.2, so we can try that here now.

Thanks; I just pushed an update.  CI will still fail due to the `resource-floats` triggering the name clash in the host bindings, which I believe you're working on.

**Comment by guybedford** [comment]
Yeah, that fix will be coming in the next major which will include the WIT updates, so I wanted to get out the patch in the mean time.

**Comment by guybedford** [comment]
jco@0.12.3 should fix the remaining resource generation bugs here now.

**Comment by dicej** [comment]
Just updated to `jco` 0.12.3.  The `resource-floats` test is fixed, but the `resource-alias-redux` test is failing now, unfortunately.  It seems that the generated host bindings for the `test` function exported by the component are wrong, but I think the code that `jco` 0.12.2 was _also_ wrong (just in a way that coincidentally allowed the test to pass).

**Comment by guybedford** [comment]
@dicej I've pushed an update to this PR for jco@0.13, with the update to the latest WITs for WASI as well. I believe all tests should be passing now.

**Pull request merged**